tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -577,6 +585,7 @@ class DxcCompiler : public IDxcCompiler3,
 #ifdef SUPPORT_QUERY_GIT_COMMIT_INFO
       ,IDxcVersionInfo2
 #endif // SUPPORT_QUERY_GIT_COMMIT_INFO
+      ,IDxcVersionInfo3
```

Comment:
I'm not sure what the point is of disallowing QI for IDxcVersionInfo2, when IDxcVersionInfo3 derives from it, and the interface is in fact now implemented.  If the point is to keep code that looks for that version of the interface from using it when it's not really supported, then perhaps IDxcVersionInfo3 should not derive from IDxcVersionInfo2.  Perhaps it should just be an independent interface with the extra method.  Otherwise one might assume that if an object implements 3, it can be assumed that it implements 2 because of the derivation in the interface definition.

The alternative seems to be to say that 2 is always implemented, but sometimes there just isn't any commit hash version info from the object, in which case we return a failure from the method, which could just be E_FAIL rather than E_NOTIMPL, since the interface is implemented, but there just isn't any hash or commit count.  The downside is that code that uses 2 might assume that when implemented, it will get valid commit info, without checking for failure, which could trigger a potential regression, but that would be bad practice by the user this interface of course.  Of course the test would have to be updated to accept this possibility too.